python/lib/dependabot/python/file_parser.rb

```diff
@@ -72,12 +72,6 @@ def poetry_dependencies
       def requirement_dependencies
         dependencies = DependencySet.new
         parsed_requirement_files.each do |dep|
-          # This isn't ideal, but currently the FileUpdater won't update
```

Comment:
I wonder what has changed to make this statement no longer hold true, maybe we should add a test to the FileUpdater that verifies this keeps working in the future?

Reply:
I added two tests. 

[One](https://github.com/dependabot/dependabot-core/pull/4969/files#diff-92c4c5a8f4322dade1159b654421d41742e07913bcb3b07db7c065fc48363f00R1328-R1346) shows that when the lockfile + requirement.txt files share a dependency what the DependencySet combiner does.  Since there's only one requirement, the rest of the update will be standard and is already tested in other specs.

[The next test](https://github.com/dependabot/dependabot-core/pull/4969/files#diff-8384d4d3db066e463df0d4c95a00929220ee4f27ee7a97d2858d6dfa3e3501d5R171-R218) shows that having two requirements is ok with the FileUpdater. It doesn't replace the dependency in both files, but it didn't before this change either! 😄